src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T item;
+        private Node<T> next;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = (Node<T>) new Node<>(null, value, null);
+            tail = head;
+        }
+        if (size > 0) {
+            Node<T> tempNode = tail;
+            tail = (Node<T>) new Node<>(tempNode, value, null);
+            tempNode.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
too complex logic. There are two cases in here, after check for IndexOutOfBoundsException:
1. If index == 0 then you should use add method
2. If an index is valid then you should find the current node by this index and change on the passed one, shifting the current one to the right

Reply:
I have done it but add method doesn't work when index == 0 but size is not empty. I need to find another solution